### PR TITLE
Update CredentialPack to use ParsedCredential

### PR DESCRIPTION
**Pull request by cobward**
Also remove `BaseCredential`, `Mdoc` and `W3CVC` types, and updated the demo app.

### CODE REVIEW

**Comment by cobward** [comment]
@Juliano1612 these `.idea` file changes happened automatically when I opened the project in android-studio. Should I revert the changes and add the files to gitignore?

**Comment by cobward** [comment]
Converted back to draft, IDE didn't immediately notify me of all the compilation errors.

**Review by Juliano1612 — APPROVED** [review]
LGTM!

File: MobileSdk/src/main/java/com/spruceid/mobile/sdk/Credential.kt [hunk 1]
```diff
@@ -32,14 +33,21 @@ private fun Mdoc.jsonEncodedDetailsInternal(elementIdentifiers: List<String>?):
...
                     }
-                    return@map Pair(id, json)
+
+                    try {
+                        val jsonArray = JSONArray(jsonString)
+                        return@map Pair(id, jsonArray)
+                    } catch (e: Exception) {
+                        print("failed to decode '$id' as JSON: $e")
+                        return@map Pair(id, jsonString)
+                    }
```
Comment (cobward) [inline MobileSdk/src/main/java/com/spruceid/mobile/sdk/Credential.kt:50]: Ah this is a really good point, most of the time the values will be JSON Strings and Integers, rather than Arrays or Objects. I'll look into how to parse this properly.

File: example/src/main/java/com/spruceid/mobilesdkexample/wallet/GenericCredentialListItem.kt [hunk 2]
```diff
@@ -159,18 +160,14 @@ fun GenericCredentialDetailsItem(credentialPack: CredentialPack) {
...
                     "credentialSubject.driversLicense.birth_date",
                 ),
-                formatter = {values ->
-                    val w3cvc = values.toList()
-                        .first { (key, _) ->
-                            val credential = credentialPack.getCredentialById(key)
-                            (credential?.asJwtVc() != null || credential?.asJsonVc() != null)
-                        }.second
+                formatter = { values ->
+                    val w3cvc = values.toList().first().second
```
Comment (cobward) [inline example/src/main/java/com/spruceid/mobilesdkexample/wallet/GenericCredentialListItem.kt:164]: @Juliano1612 can I revert this change? It's also possible for the value to be an mDoc, in which case `w3cvc["credentialSubject"]` will always be null

File: example/src/main/java/com/spruceid/mobilesdkexample/wallet/GenericCredentialListItem.kt [hunk 3]
```diff
@@ -159,18 +160,14 @@ fun GenericCredentialDetailsItem(credentialPack: CredentialPack) {
...
 
                     var portrait = ""
                     var firstName = ""
                     var lastName = ""
                     var birthDate = ""
-                    if(w3cvc["credentialSubject.driversLicense"].toString().isNotEmpty()) {
+                    if (w3cvc["credentialSubject.driversLicense"].toString().isNotEmpty()) {
```
Comment (cobward) [inline example/src/main/java/com/spruceid/mobilesdkexample/wallet/GenericCredentialListItem.kt:170]: Does the `keyPathFinder` function need to be used here instead of indexing?

File: example/src/main/java/com/spruceid/mobilesdkexample/wallet/GenericCredentialListItem.kt (hunk 2 above)
Comment (Juliano1612) [reply example/src/main/java/com/spruceid/mobilesdkexample/wallet/GenericCredentialListItem.kt:164]: @cobward, you can revert, but @w4ll3 and I are completely refactoring this component. I updated it to keep the app working, but I wouldn't worry about that now.

File: example/src/main/java/com/spruceid/mobilesdkexample/wallet/GenericCredentialListItem.kt (hunk 3 above)
Comment (Juliano1612) [reply example/src/main/java/com/spruceid/mobilesdkexample/wallet/GenericCredentialListItem.kt:170]: The `keyPathFinder` was incorporated when searching for claims. So, if I have an object in the format

```json
{
  "issuer": {
    "address": {
      "number": 123
    }
  }
}
```
I'll ask for the `issuer.address.number` property, and the formatter returns the object

```json
{
  "issuer.address.number": 123
}
```

That's why I can do index access this way

**Review by Ryanmtate — APPROVED** [review]
I think this is good to merge. 

I will add the `addSdJwt` method for the cred pack in a follow up PR.
